wasm/jit/jit_arm64_test.go

```diff
@@ -1743,3 +1746,277 @@ func TestArm64Compiler_compieleCall(t *testing.T) {
+								}
+								require.NoError(t, err)
+							}
+
+							// Push the selection seed.
+							if selectX1 {
+								err = compiler.compileConstI32(&wazeroir.OperationConstI32{Value: 1})
+							} else {
+								err = compiler.compileConstI32(&wazeroir.OperationConstI32{Value: 0})
+							}
+							require.NoError(t, err)
+
+							err = compiler.compileSelect()
+							require.NoError(t, err)
+
+							err = compiler.compileReturnFunction()
+							require.NoError(t, err)
+
+							code, _, _, err := compiler.compile()
+							require.NoError(t, err)
+
+							env.exec(code)
+							require.Equal(t, jitCallStatusCodeReturned, env.jitStatus())
+
+							// Check if the correct value is chosen.
+							if selectX1 {
+								require.Equal(t, vals[0], env.stackTopAsUint64())
+							} else {
+								require.Equal(t, vals[1], env.stackTopAsUint64())
+							}
+						})
+					}
+				})
+			}
+		})
+	}
+}
+
+func TestArm64Compiler_compieleSwap(t *testing.T) {
+	const x, y uint64 = 100, 200
+	op := &wazeroir.OperationSwap{Depth: 10}
+
+	env := newJITEnvironment()
+	compiler := env.requireNewCompiler(t)
+	err := compiler.compilePreamble()
+	require.NoError(t, err)
+
+	// Setup the initial values on the stack would look like: [y, ...., x]
+	err = compiler.compileConstI64(&wazeroir.OperationConstI64{Value: y})
+	require.NoError(t, err)
+	// Push the middle dummy values.
+	for i := 0; i < op.Depth-1; i++ {
+		compiler.locationStack.pushValueLocationOnStack()
+	}
+	err = compiler.compileConstI64(&wazeroir.OperationConstI64{Value: x})
+	require.NoError(t, err)
+
+	err = compiler.compileSwap(op)
+	require.NoError(t, err)
+
+	// After the swap, both values must be on registers.
+	require.True(t, compiler.locationStack.peek().onRegister())
+	require.True(t, compiler.locationStack.stack[0].onRegister())
+
+	err = compiler.compileReturnFunction()
+	require.NoError(t, err)
+
+	// Generate the code under test.
+	code, _, _, err := compiler.compile()
+	require.NoError(t, err)
+
+	// Run code.
+	env.exec(code)
+
+	require.Equal(t, uint64(op.Depth+1), env.stackPointer())
+	// y must be on the top due to Swap.
+	require.Equal(t, y, env.stackTopAsUint64())
+	// x must be on the bottom.
+	require.Equal(t, x, env.stack()[0])
+}
+
+func TestAmd64Compiler_initializeModuleContext(t *testing.T) {
+	for _, tc := range []struct {
+		name           string
+		moduleInstance *wasm.ModuleInstance
+	}{
+		{
+			name: "no nil",
+			moduleInstance: &wasm.ModuleInstance{
+				Globals: []*wasm.GlobalInstance{{Val: 100}},
+				// TODO: Add memory, table
+			},
+		},
+		// TODO: Add memory, table
+		{
+			name:           "nil",
+			moduleInstance: &wasm.ModuleInstance{},
+		},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			env := newJITEnvironment()
+			compiler := env.requireNewCompiler(t)
+			compiler.f.ModuleInstance = tc.moduleInstance
+
+			// The assembler skips the first instruction so we intentionally add NOP here.
```

Comment:
any note to add about why it does that?

Reply:
I have no idea for the reason but I just confirmed from the code (in golang-asm).. this will become unnecessary after implementing our assembler :D